src/main/java/com/sphy/hotelmanagementapplication/service/RoomService.java

```diff
@@ -91,94 +92,112 @@ public List<RoomDTO> saveRooms(List<RoomDTO> roomsDTO) throws Exception{
         for (Room room:rooms){
             roomDTOS.add(roomToRoomDTO.converter(room));
         }
-
-
         return roomDTOS;
     }
 
     /***
      * get all rooms
      * @return a list of all rooms
-     * @throws Exception
+     * @throws ApiRequestException if no room is saved
      */
-    public List<RoomDTO> getRooms() throws Exception {
-		List<Room> rooms = new ArrayList<>();
-        List<RoomDTO> roomsDTO = new ArrayList<>();
-        roomRepository.findAll().forEach(rooms::add);
-        for (Room room : rooms){
-            roomsDTO.add(roomToRoomDTO.converter(room));
+    public List<RoomDTO> getRooms() throws ApiRequestException {
+
+        if (roomRepository.findAll() == null) {
```

Comment:
There is no need to add his exception here, because in react, i ll check if the array is empty and i ll show different things in each case